text/0000-system-buildpacks-in-builder.md

```diff
@@ -0,0 +1,145 @@
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Unless otherwise stated, system buildpacks conform to the [buildpack API](https://github.com/buildpacks/spec/blob/main/buildpack.md).
+
+The `system.*pre*.buildpacks` will be provided to the lifecycle into a new file, `system.toml`. The `[system]` table in `system.toml` will be processed by the lifecycle, and each `pre`/`post` buildpack will run during the detect phase. Those that pass detection will run during the build phase.
+
+## Detection
+
+The exit code of detection by system buildpacks MUST influence the selected buildpack group. If a system buildpack is non-optional and fails detection, then detection MUST for that group fail. If a system buildpack is optional and passes detection, then detection MAY pass for that group.
+
+System buildpacks may require/provide in the build plan following standard buildpack API specification.
+
+## Build
+
+System buildpacks that have passed detection will be added to `group.toml` and treated like any other buildpack for the remainder of the build.
+
+If a system buildpack exits with a status of `100`, the build will fail.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Default system buildpacks are hidden from the user before the build and their execution may be unexpected.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+
+End users would have to add buildpacks like the `profile-buildpack` or other buildpacks that implement system/spec behaviors themselves.
+
+## Use the `[[order]]` table
+
+Instead of a new `[system]` table, we could put `pre` and `post` in the `[[order]]` table. However, this could imply that there is a interaction/override/etc between these buildpacks and the `pre`/`post` buildpacks in `project.toml`. But there is not.
+
+## Use the `[lifecycle]` table
+
+```
+[lifecycle]
+  version = "<string>"
+
+  [[lifecycle.pre.buildpacks]]
+    id = "<buildpack ID>"
+    version = "<buildpack version>"
+    optional = false
+
+  [[lifecycle.post.buildpacks]]
+    id = "<buildpack ID>"
+    version = "<buildpack version>"
+    optional = false
+```
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [RFC-0078: Group additions to Builder order](https://github.com/buildpacks/rfcs/blob/main/text/0078-group-additions.md)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Should the system buildpacks that pass detection be added to the buildpack group? This might make it difficult to hide them.
+    - disposition: yes. hiding is optional and up to the platform
+- Should the system buildpacks output to stdout/stderr and should that be displayed?
+    - disposition: it may be
```

Comment:
```suggestion
```
Given that we have agreed not to do anything "special" beyond adding these buildpacks to the order I think this is also resolved. The system buildpacks will print to stdout/stderr just like all the other buildpacks.